lib/action_network_rest/base.rb

```diff
@@ -17,6 +17,49 @@ def list(page: 1)
       objects
     end
 
+    def all
+      min_wait = 0.25 # sec to wait between calls
+      timestamp = Time.now.to_r # secs since Unix Epoch
+      page_number = 1
+      tries = 0
+      all = []
+      loop do
+        # Limit rate before request to avoid error
+        wait = min_wait * (1.5**tries)
+        new_time = Time.now.to_r
+        if (timestamp + min_wait - new_time).positive?
+          sleep(wait) # Wait if calling again could excede the rate limit
+          tries += 1
+          next # check again
+        end
+
+        begin
+          page = list(page: page_number)
+        rescue ActionNetworkRest::Response::TooManyRequests
+          if tries >= 10
+            raise ActionNetworkRest::Response::UsedAllRequestTries,
+                  {
+                    tries: tries,
+                    last_wait_length: wait,
+                    resource: base_path
+                  }.to_json
+          end
+
+          sleep(wait)
+          tries += 1 # Exponential back off if got Too Many Requests error reponse
+          next
+        end
+        timestamp = Time.now.to_i
```

Comment:
This probably works fine, but just for consistency should it be?
```suggestion
        timestamp = Time.now.to_r
```